ci-operator/step-registry/cucushift/installer/check/ibmcloud/custom-endpoints/cucushift-installer-check-ibmcloud-custom-endpoints-commands.sh

```diff
@@ -28,11 +15,100 @@ function get_service_endpoint() {
     echo "${service_endpoint}"
 }
 
+function exists_in()
+{
+	local val=$1
+	shift
+	local val_list=("$@")
+	local match=1
+	for v in "${val_list[@]}";
+	do
+	    if [[ "$val" = "$v" ]]; then
+	        match=0
+	        break
+	    fi
+	done
+	return $match
+}
+
+function isPreVersion() {
+  local REQUIRED_OCP_VERSION="$1"
+  local isPre
+  version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1,2)
+  echo "get ocp version: ${version}"
+
+  isPre=0
+  if [ -n "${version}" ] && [ "$(printf '%s\n' "${REQUIRED_OCP_VERSION}" "${version}" | sort --version-sort | head -n1)" = "${REQUIRED_OCP_VERSION}" ]; then
+    isPre=1
+  fi
+  return $isPre
+}
+
+function check_ep_names() {
+    local isPre418=$1
+    local output expString
+    output=$(openshift-install explain installconfig.platform.ibmcloud.serviceEndpoints)
+    if isPre418; then
+        expString='Valid Values: "COS","DNSServices","GlobalCatalog","GlobalSearch","GlobalTagging","IAM","ResourceController","ResourceManager","VPC"'
+    else
+        expString='Valid Values: "CIS","COS","COSConfig","DNSServices","GlobalCatalog","GlobalSearch","GlobalTagging","HyperProtect","IAM","KeyProtect","ResourceController","ResourceManager","VPC"'
+    fi
+    if ! [[ ${output} == *"${expString}"* ]]; then
+        echo "ERROR: Unexpected explain installconfig.platform.ibmcloud.serviceEndpoints: - ${output} "
+        return 1
+    else
+        echo "Check installconfig.platform.ibmcloud.serviceEndpoints passed."
+        return 0
+    fi
+}
+
+function check_eps() {
+    local isPre418=$1
+    local epFile url ret srv_418 src_name service_endpoint url
+    ret=0
+    epFile="${ARTIFACT_DIR}/eps.json"
+    oc get infrastructure -o json | jq -r '.items[0].status.platformStatus.ibmcloud.serviceEndpoints' > ${epFile}
+    declare -a srv_array
+    readarray -t srv_array < <(jq -r 'keys[]' ${DEFAULT_PRIVATE_ENDPOINTS})
+    srv_418=("COSConfig" "GlobalCatalog" "KeyProtect")
+    for srv in "${srv_array[@]}"; do
+        src_name="SERVICE_ENDPOINT_${srv}"
+        if [[ -n "${!src_name}" ]]; then
+            service_endpoint=$(get_service_endpoint "${srv}" "${!src_name}")
+            url=$(cat ${epFile} | jq -r --arg n ${srv} '.[] | select (.name==$n) | .url')
+            echo "${srv} service_endpoint: [${service_endpoint}] url: [${url}] from the infrastructure."
+            if [ -n "$url" ] && [ -n "$service_endpoint" ]; then
+                if [[ "$url" != "$service_endpoint" ]]; then
+                    echo "ERROR 1: [${srv}] endpoint expected is [${service_endpoint}], but get url [${url}] from the infrastructure!!"
+                    ret=$((ret + 1))
+                else
+                    echo "${srv} custom endpoint check passed"
+                fi   
+            elif [[ -z "$url" ]] && [[ $isPre418 -eq 0 ]] && exists_in ${srv} "${srv_418[@]}"; then
```

Comment:
Actually you do not need some complicated logic to handle version difference, you just to prepare target service array list for different versions, just like https://github.com/openshift/release/pull/58984/files#r1859767876, only need two lines.